Data/Set/Ordered.hs

```diff
@@ -36,6 +36,8 @@ import qualified Data.Map as M
 data OSet a = OSet !(Map a Tag) !(Map Tag a)
 
 -- | Values appear in insertion order, not ascending order.
+instance Ord a => Semigroup (OSet a) where (<>) = foldl' (|>)
```

Comment:
It might be a bit more efficient to use `(|<>)` here.

Reply:
Were we to move forward with this change, I agree that it would make sense to implement `<>` in terms of `|<>`, even if just for the sake of the symmetry. I've looked at the implementation of `|<>` and haven't yet figured out exactly what its complexity is, but it _is_ likely more efficient than the fold proposed here.